Cannot confirm a user invite to an organization when using email alias

How to reproduce:

1. Invite a new user with an email alias (user+alias@domain.tld) to an organization
2. The new user follows the links in the invite email
3. The new user creates an account. The email address is prefilled in the form, but the '+' from the email address is missing. Create the account without adding the '+' back.
4. Log in as an admin
5. Accept the newly invited user
6. The site now display an error: "Data provided to an operation does not meet requirements"

I'm not seeing any errors in the server logs, apart from these:

```
[2019-04-29 13:33:17][rocket::rocket][INFO] GET /undefined image/webp:                                                                     
[2019-04-29 13:33:17][_][INFO] Matched: GET /<p..> [10] (web_files)                                                                        
[2019-04-29 13:33:17][_][ERROR] Response was a non-`Responder` `Err`: Os { code: 2, kind: NotFound, message: "No such file or directory" }.
[2019-04-29 13:33:17][_][INFO] Outcome: Failure                                                                                            
[2019-04-29 13:33:17][_][WARN] Responding with 500 Internal Server Error catcher. 
```
.. which do not seem to be related to the issue.

I'm running bitwarden_rs from the docker image with a Caddy server and https.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -105,6 +105,9 @@ handlebars = "1.1.0"
 soup = "0.4.1"
 regex = "1.1.7"
 
+# URL encoding library
+percent-encoding = "1.0.1"
+
 [patch.crates-io]
 # Add support for Timestamp type
 rmp = { git = 'https://github.com/dani-garcia/msgpack-rust' }

--- a/src/mail.rs
+++ b/src/mail.rs
@@ -4,6 +4,7 @@ use lettre::{ClientSecurity, ClientTlsParameters, SmtpClient, SmtpTransport, Tra
 use lettre_email::{EmailBuilder, MimeMultipartType, PartBuilder};
 use native_tls::{Protocol, TlsConnector};
 use quoted_printable::encode_to_str;
+use percent_encoding::{percent_encode, DEFAULT_ENCODE_SET};
 
 use crate::api::EmptyResult;
 use crate::auth::{encode_jwt, generate_invite_claims};
@@ -101,7 +102,7 @@ pub fn send_invite(
             "url": CONFIG.domain(),
             "org_id": org_id.unwrap_or_else(|| "_".to_string()),
             "org_user_id": org_user_id.unwrap_or_else(|| "_".to_string()),
-            "email": address,
+            "email": percent_encode(address.as_bytes(), DEFAULT_ENCODE_SET).to_string().replace("+", "%2b"),
             "org_name": org_name,
             "token": invite_token,
         }),

